waspc/data/Generator/templates/sdk/wasp/server/utils.ts

```diff
@@ -19,14 +19,14 @@ type RequestWithExtraFields = Request & {
  *   if given middleware returns promise, reject of that promise will be correctly handled,
  *   meaning that error will be forwarded to next().
  */
-export const handleRejection = (
+export const handleRejection = <Req extends Request = RequestWithExtraFields, Res extends Response = Response>(
   middleware: (
-    req: RequestWithExtraFields,
-    res: Response,
+    req: Req,
+    res: Res,
     next: NextFunction
   ) => any
 ) =>
-async (req: RequestWithExtraFields, res: Response, next: NextFunction) => {
+async (req: Req, res: Res, next: NextFunction) => {
   try {
     await middleware(req, res, next)
```

Comment:
The linter throws a warning for this `await`:

![image](https://github.com/user-attachments/assets/ee748137-05df-45b8-85a5-34537441a35b)

Best to comment on why it's necessary, like we did [here](https://github.com/wasp-lang/wasp/blob/79992e9f289a50d3a670965695351548afe10db0/waspc/data/Generator/templates/sdk/wasp/client/operations/actions/core.ts#L19).

Reply:
Outdated code since https://github.com/wasp-lang/wasp/pull/2685